### DISCUSSION TITLE
How to capture a pair with query?

### DISCUSSION THREAD

**Discussion by ghost** [tree-sitter/tree-sitter#1781] [Q&A] [answered]
Given a Go snippet:
```go
var x,y = 1,2;
```
with its sep:
```
(source_file [0, 0] - [12, 0]
  (var_declaration [0, 0] - [0, 15]
    (var_spec [0, 4] - [0, 15]
      name: (identifier [0, 4] - [0, 5])
      name: (identifier [0, 7] - [0, 8])
      value: (expression_list [0, 11] - [0, 15]
        (int_literal [0, 11] - [0, 12])
        (int_literal [0, 14] - [0, 15]))))
```
I tried to query with:
```
(var_spec name: (_) @var value: (_) @val)
```
but always get:
![image](https://user-images.githubusercontent.com/57096917/174715770-94a0ccbf-5bf8-46cc-9c1c-c38de27ec720.png)

how can I capture the var declaration as (x, 1) , (y, 2)?

**Comment by ris58h** [comment, marked as answer]
Hi! I'm not a tree-sitter expert, but seems like there is an issue with querying multiple fields with the same name.

You could use
```
(var_spec
  (identifier) @var
  value: (expression_list (_) @val))
```
to get all the vars and all the vals in one query and hope that the order will be the same.

Or you could use multiple queries
```
(var_spec
  . (identifier) @var1
  value: (expression_list . (_) @val1))

(var_spec
  . (identifier) . (identifier) @var2 
  value: (expression_list . (_) . (_) @val2))
  
(var_spec
  . (identifier) . (identifier) . (identifier) @var3
  value: (expression_list . (_) . (_) . (_) @val3))
```
on query-per-pair basis.

Hope this will help.

**Reply by ghost** [reply]
Really thanks :)
